### ISSUE TITLE
Separate tabs in code editor

### ISSUE THREAD

**Issue by trelemar** [nesbox/TIC-80#569]
The idea is you could keep your code better organized with different tabs to switch between within the editor. Another solution MonstersGoBoom had was to add "tags" users could easily navigate to.

Fubuki, MGB, and myself all think this would be very useful.

**Comment by hseiken** [comment]
Any options that allow quick jumping to parts of code is a good idea, implementation isn't really a concern, as long as it's within a key or mouse click to jump to functions and such.  

+1 for this idea.

**Comment by ambyra** [comment]
How does the bank switching work currently? Can a bank access code from another bank? Can variables store functions from other banks?

**Comment by StinkerB06** [comment]
Yeah, in the latest PICO-8 version, there are tabs at the top in the code editor.

**Comment by AnastasiaDunbar** [comment]
Other than Lua, will it support for JavaScript too?

**Comment by carlsmith** [comment]
Support for MoonScript, JavaScript, Wren and Fennel [has been added already][1].

[1]: https://github.com/nesbox/TIC-80/wiki#cartridge-metadata

**Comment by AnastasiaDunbar** [comment]
I meant if these tabs would work the same for other languages (if all of them are executed).

**Comment by AnastasiaDunbar** [comment]
When this gets added, can we put the shader in a tab instead of a bank if it's possible? (And also, again, if all of the tabs aren't executed on runtime.)

**Comment by nesbox** [comment]
I'm thinking to remove banks for code and make it works like tabs, because it generates a lot of problems with error handling, code loading and etc.
And yes, we can put shader to the second tab in this case.

**Comment by ambyra** [comment]
Great idea! In Qbasic for DOS, you could put new functions in their own tab. So instead of tab numbers (1,2,3,4) you could have a drop down box with assignable function names (main, TVshader, physics, animation, etc). Tab 1 would be hardcoded with name Main, and must have TIC() function in it. Tab 2 would be named Shader, must have screen shader. The rest would take the name of the outermost function that was put in them.

**Comment by carlsmith** [comment]
Just a reminder: Any solution has to work with the supported languages, and ideally, future language additions.

**Comment by StinkerB06** [comment]
@nesbox Were you concatenating all of the code banks together with a newline separating each bank?

It would've been better if you implemented each code bank as 8 text banks concatenated together before compiling the code.

**Comment by RaspberyPye** [comment]
Any update on this?

**Comment by nesbox** [comment]
not yet, only bookmarks have been added in the latest version :)

**Comment by sedyh** [comment]
Hello, is there any update on this?
